### ISSUE TITLE
Converted fonts produce gibberish in Allegro

### ISSUE THREAD

**Issue by propapanda** [SuperIlu/GrxFntConv#1]
Very interesting tool, but sadly I can't get it to work properly. I compiled under Linux and then converted some random .TTF files I had on my system. Some produce warnings, others don't, but in the end I always get a .FNT file that, at least from what I can tell, has reasonable content.

The .FNTs all load fine in Allegro 4.2.2, but unfortunately the text is then rendered as gibberish, i.e.:
![allegro](https://user-images.githubusercontent.com/31049568/144767673-36f1761f-52e0-4ada-8ff8-3ea091f61a86.gif)

Tried probably two dozen .TTFs, also at varying sizes, and so far I have yet to produce a single one that actually renders properly. On the other hand, the .FNT files supplied with DOjS all render just fine for me in Allegro, so I don't think it's an issue on the loading side...

Any ideas?

**Comment by SuperIlu** [comment]
As far as I remember only fixed width TTF are supported. I have to dig into the code again to check.
I only tested it with the fonts from DamienG https://damieng.com/typography/zx-origins/ and maybe one or two console fonts.
I did not spend more work on TTF after Damien suggested to use BDF and be done with rendering artifacts :)

Did you try any TTF font that is legally available for download somewhere for me to check?

**Comment by propapanda** [comment]
Thanks for looking into it.

I think I figured it out: The problem only occurs when creating ASCII .FNTs via the "c" or "f" flags. Apparently Allegro has trouble decoding these properly. Simply running fntconv without the flags produces a binary .FNT and this happens to render perfectly fine in Allegro. So far it's worked with any TTF I've tried, even those with variable widths.

If you want to reproduce the issue: The fonts you linked from DamienG produce the same gibberish when converted to ASCII (e.g. "Hourglass", both TTF and BDF). 
Then again, the workaround using binary fonts is simple enough and I'm not even sure Allegro is expected to work with the ASCII ones... so we can probably close this issue.

Thanks again!
